### PR TITLE
Search courses features

### PR DESCRIPTION
**Pull request by vandonfx**

### CODE REVIEW

**Comment by EndureBlackout** [comment]
@wallerb7039 It looks like Brandon's code for searching courses got removed from your branch somehow. Did you merge staging and pop your changes on top of that?

**Review by EndureBlackout — APPROVED** [review]

**Comment by wallerb7039** [comment]
Yeah, I wanted to get all of the tests for delete course, I didn’t even consider the search course side of things.



________________________________

**Review by wallerb7039 — APPROVED** [review]

File: routes/courses.js [hunk 1]
```diff
@@ -8,6 +9,35 @@ const { authorizeSession } = require('./../services/auth');
...
+      return res.send(courses)
+    } catch (error) {
+      next(error);
+    }
+  })
+
+  router.get('/:courseid', authorizeSession, async (req, res, next) => {
+    try {
+      const courseid = req.params.courseid;
+      const courses = await Course.findAll({courseid: courseid});
```
Comment (EndureBlackout) [inline routes/courses.js:30]: This should be "id" not "courseid"

File: routes/courses.test.js [hunk 2]
```diff
@@ -7,11 +7,26 @@ const User = require('../models/User');
...
   log.disableAll();
 });
+// use for creating course 
+function dataForGetCourses(rows, offset = 0) {
+  const data = [];
+  for (let i = 1; i <= rows; i++) {
+    const value = i + offset;
+    data.push({
+      id: `${value}`,
+      courseid: value ,
```
Comment (EndureBlackout) [inline routes/courses.test.js:17]: This should reflect the changes to the database from "courseid" to section

**Comment by EndureBlackout** [comment]
@vandonfx Left some comments where changes are needed.

**Review by EndureBlackout — APPROVED** [review]

**Pull request merged**